pkg/apis/infinispan/v2alpha1/restore_types.go

```diff
@@ -0,0 +1,91 @@
+package v2alpha1
+
+import (
+	v1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BackupSpec defines the desired state of Backup
+type RestoreSpec struct {
+	Cluster   string                     `json:"cluster"`
+	Backup    string                     `json:"backup"`
+	Resources RestoreResources           `json:"resources,optional,omitempty"`
+	Container v1.InfinispanContainerSpec `json:"container,optional,omitempty"`
+}
+
+type RestoreResources struct {
+	Caches       []string `json:"caches,optional,omitempty"`
+	CacheConfigs []string `json:"cacheConfigs,optional,omitempty"`
+	Counters     []string `json:"counters,optional,omitempty"`
+	ProtoSchemas []string `json:"protoSchemas,optional,omitempty"`
+	Scripts      []string `json:"scripts,optional,omitempty"`
+}
+
+type RestorePhase string
+
+const (
+	// BackupInitializing means the request has been accepted by the system, but the underlying resources are still
+	// being initialized.
+	RestoreInitializing RestorePhase = "Initializing"
+	// RestoreInitialized means that all required resources have been initialized.
+	RestoreInitialized RestorePhase = "Initialized"
+	// RestoreRunning means that the Restore pod has been created and the Restore process initiated on the infinispan server.
+	RestoreRunning RestorePhase = "Running"
+	// RestoreSucceeded means that the Restore process on the server has completed and the Restore pod has been terminated.
+	RestoreSucceeded RestorePhase = "Succeeded"
+	// RestoreFailed means that the Restore failed on the infinispan server and the Restore pod has terminated.
+	RestoreFailed RestorePhase = "Failed"
+	// RestoreUnknown means that for some reason the state of the Restore could not be obtained, typically due
+	// to an error in communicating with the underlying Restore pod.
+	RestoreUnknown RestorePhase = "Unknown"
+)
+
+// RestoreStatus defines the observed state of Restore
+type RestoreStatus struct {
+	// State indicates the current state of the restore operation
+	Phase RestorePhase `json:"phase"`
```

Comment:
@ryanemerson we should add `Conditions` in additional to `Phase` implementation. Phase is easy for operator itself but not useful for testing and customer watching as it can be changing independently and never can't see already processed phase outside operator.
Alternative way: some functions like `IsInitializedOfFailed`, `IsRunningOrFailed`, etc. can be implemented on top of the Backup/Restore structs and they will validate that intermediate or end phase already processed by operator. It can be possible to use this functions internally or just for testing only.